pkg/network/httpx/server.go

```diff
@@ -0,0 +1,136 @@
+package httpx
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/giongto35/cloud-game/v2/pkg/service"
+	"golang.org/x/crypto/acme/autocert"
+)
+
+type Server struct {
+	http.Server
+	service.RunnableService
+
+	autoCert *autocert.Manager
+	opts     Options
+
+	listener *Listener
+	redirect *Server
+}
+
+func NewServer(address string, handler func(serv *Server) http.Handler, options ...Option) (*Server, error) {
+	opts := &Options{
+		Https:         false,
+		HttpsRedirect: true,
+		IdleTimeout:   120 * time.Second,
+		ReadTimeout:   5 * time.Second,
+		WriteTimeout:  5 * time.Second,
+	}
+	opts.override(options...)
+
+	server := &Server{
+		Server: http.Server{
+			Addr:         address,
+			IdleTimeout:  opts.IdleTimeout,
+			ReadTimeout:  opts.ReadTimeout,
+			WriteTimeout: opts.WriteTimeout,
+		},
+		opts: *opts,
+	}
+	// (╯°□°)╯︵ ┻━┻
```

Comment:
Because of this dubious part: https://github.com/giongto35/cloud-game/blob/f28e766da7be3f4bf581b4befb5dc3e9920c1f84/pkg/network/httpx/server.go#L128-L131

I have to pass the server instance into the external HTTP handler definition callback.